### PR TITLE
udev: cleanups for event blocker

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/udev/udevd.c [hunk 1]
```diff
@@ -854,9 +854,21 @@ static int event_run(Event *event) {
...
+
+        if (!a || !b)
+                return false;
+
+        for (; *a != '\0' && *b != '\0'; a++, b++)
+                if (*a != *b)
+                        return false;
+
+        return *a == '/' || *b == '/' || *a == *b;
+}
```
Comment (poettering) [inline src/udev/udevd.c]: could use a test case I think. given this is not obvious

Comment (poettering) [reply src/udev/udevd.c]: hmm, *a == *b only holds if both are NUL, right?

**Comment by poettering** [comment]
PR looks good to me, but the wider implications are not entirely clear to me, hence: stuff for v251? or v252? what do you think?

File: src/udev/udevd.c (hunk 1 above)
Comment (yuwata) [reply src/udev/udevd.c]: > hmm, *a == *b only holds if both are NUL, right?

Right.

**Comment by yuwata** [comment]
> PR looks good to me, but the wider implications are not entirely clear to me, hence: stuff for v251? or v252? what do you think?

I'd like this to be in v251 with other PRs for udev. Especially, the third commit should be important to make inotify watch stabler.

File: src/udev/udevd.c (hunk 1 above)
Comment (yuwata) [reply src/udev/udevd.c]: Added testcases.

**Comment by keszybz** [comment]
LGTM.

**Pull request merged**